### PR TITLE
Granular admin perms

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Allows an admin rank to be split into multiple ranks--e.g. `Game Master+Coder`. Will combine the rights together.

**You can no longer edit rank permissions, temporarily or permanently, from the permissions panel. Please use the admin_ranks database table or admin_ranks.txt instead.**

TODO:
- [x] New admin ranks panel for DB
- [x] Remove dbconfig thing I always forget it
- [x] Remove other config changes I made here

## Why It's Good For The Game

Codermin has +POLL.
Admin Trainer does not have +POLL.

It is my duty to fix this stupid shit.

## Changelog

:cl:
server: Admin ranks can now be combined with a plus sign (for example, Game Master+Coder). Single admin ranks with a plus sign in their name are no longer supported.
server: You can no longer edit rank permissions, temporarily or permanently, from the permissions panel. Please use the admin_ranks database table or admin_ranks.txt instead.
/:cl:

### CODE REVIEW

File: config/dbconfig.txt [hunk 1]
```diff
@@ -3,31 +3,31 @@
 ## administration, and the in game library.
 
 ## Should SQL be enabled? Uncomment to enable
-#SQL_ENABLED
+SQL_ENABLED
```
Comment (optimumtact) [inline config/dbconfig.txt:6]: crinky cronge

**Comment by LemonInTheDark** [comment]
IDED pr, malding, git gud, etc etc etc

File: config/dbconfig.txt (hunk 1 above)
Comment (Mothblocks) [reply config/dbconfig.txt:6]: the hilarious thing is this makes the server just straight up hang, probably CI as well

File: code/modules/admin/admin_ranks.dm [hunk 2]
```diff
@@ -199,6 +199,33 @@ GLOBAL_PROTECT(protected_ranks)
...
+/proc/ranks_from_rank_name(rank_name)
+	var/list/rank_names = splittext(rank_name, "+")
+	var/list/ranks = list()
+
+	for (var/datum/admin_rank/rank as anything in GLOB.admin_ranks)
+		if (rank.name in rank_names)
+			rank_names -= rank.name
+			ranks += rank
+
+			if (rank_names.len == 0)
```
Comment (Fikou) [inline code/modules/admin/admin_ranks.dm:212]: why not !length(rank_names)?

Comment (Mothblocks) [reply code/modules/admin/admin_ranks.dm:212]: I don't like treating non-bool 0 as falsy unless it happens to help in some other way (such as also testing empty strings/false at the same time).

Comment (LemonInTheDark) [reply code/modules/admin/admin_ranks.dm:212]: ^ cringe

**Comment by Mothblocks** [comment]
I'm still working on this but the work left is so fucking boring

**Comment by Mothblocks** [comment]
> mothblocks: how important is the ability to create temporary roles and such in game? and also i think it lets you edit dbranks too? the existing stuff in permission panel, but NOT setting peoples role, but CREATING them. it is a lot of boring work to keep with my pr and i want to know if i can laze out

> mso: i want there to be the ability to add flags to existing people temporally, nothing more.

**Comment by MrStonedOne** [comment]
For downstream purposes you need to document that you've removed the rank editing code from the permission panel in big bold letters in the pr as well as a `server:` tag in the changelog.

for full context: What /tg/ does with admin configs + dbrank, is the result of enabling three default disabled configs. one to treat admins.txt as super admins over dbadmins, one to treat admin_ranks.txt as super ranks, with standing over the dbranks, and finally one to disable editing ranks in game.

Based on default configs, the ""technically" normal or """correct""" way of doing admins is to use the admins.txt to seed the config, then blanking both admins.txt and admin_ranks.txt and only doing db config for both. /tg/ has never even attempted to do it that way, but its how it was coded.

(You can (and should) look into the db admin config(s) and think about re-working that to better fit what we've figured out to be more sane workflows for the default.)

**Comment by Mothblocks** [comment]
Added it to the changelog
